yaes-core/src/main/scala/in/rcard/yaes/Async.scala

```diff
@@ -266,6 +266,80 @@ object Async {
     def forkOn()(using async: Async): Fiber[Unit] = Async.fork {
       flow.collect { _ => () }
     }
+
+    /**
+     * Combines the elements of this flow with another flow using the provided function.
+     *
+     * The method emits elements by applying the provided function `f` to pairs of elements
+     * from the current flow and the `other` flow. It only emits elements when both flows
+     * provide values.
+     *
+     * Example:
+     * {{{
+     * val flow1 = Flow("a", "b", "c", "d")
+     * val flow2 = Flow(1, 2, 3)
+     * val combined = flow1.zipWith(flow2)((_, _))
+     *
+     * val result = scala.collection.mutable.ArrayBuffer[(String, Int)]()
+     *
+     * combined.collect { result += _ }
+     *
+     * // Result contains the elements ("a", 1), ("b", 2), ("c", 3)
+     * }}}
+     *
+     * @param other
+     * The other flow to combine with this flow.
+     * @param f
+     * A function that takes a pair of elements from both flows and produces a new element.
+     * @return
+     * A new flow emitting elements resulting from applying the function `f` to pairs
+     * of elements from both flows.
+     */
+    def zipWith[B, C](other: Flow[B])(f: (A, B) => C): Flow[C] =
+      Flow.flow {
+        Async.run {
```

Comment:
I don't want the library to run the effect in a hidden way. The end user should always call the `Async.run` method. Moreover, I want to try to implement the `zipWith` (or `zip`) using the available structured concurrency primitives. We have to understand if it's possible. 

The Kotlin implementation uses the `Channel<T>` type: https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/common/src/flow/internal/Combine.kt#L82